airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/split_secrets/VaultSecretPersistenceTest.java

```diff
@@ -0,0 +1,58 @@
+package io.airbyte.config.persistence.split_secrets;
+
+import org.apache.commons.lang3.RandomUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.vault.VaultContainer;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class VaultSecretPersistenceTest {
+  private VaultSecretPersistence persistence;
+  private String baseCoordinate;
+
+  private VaultContainer vaultContainer;
+
+  @BeforeEach
+  void setUp() {
+    vaultContainer = new VaultContainer("vault").withVaultToken("vault-dev-token-id");
+    vaultContainer.start();
+
+    final var vaultAddress = "http://" + vaultContainer.getHost() + ":" + vaultContainer.getFirstMappedPort();
+
+    persistence = new VaultSecretPersistence(vaultAddress, "secret/testing", "vault-dev-token-id");
+    baseCoordinate = "VaultSecretPersistenceIntegrationTest_coordinate_" + RandomUtils.nextInt() % 20000;
+  }
+
+  @AfterEach
+  void tearDown() {
+    vaultContainer.stop();
+  }
+
+  @Test
+  void testReadWriteUpdate() {
+    final var coordinate1 = new SecretCoordinate(baseCoordinate, 1);
+
+    // try reading non-existent value
+    final var firstRead = persistence.read(coordinate1);
+    assertTrue(firstRead.isEmpty());
```

Comment:
Nit/Not needed for this review: We have started using assertJ that provides more meaningful failure reason. this could be written like: `assertThat(firstRead).isTrue`.